packages/block-editor/src/components/link-control/use-internal-input-value.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+
+export default function useInternalInputValue( value ) {
+	const [ internalInputValue, setInternalInputValue ] = useState(
+		value || ''
+	);
+
+	useEffect( () => {
+		/**
+		 * Update the state value internalInputValue if the url value changes
+		 * for example when clicking on another anchor
+		 */
+		if ( value && value !== internalInputValue ) {
+			setInternalInputValue( value );
+		}
+	}, [ value ] );
```

Comment:
I think you can just do this (more info: https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops):
```suggestion
	if ( value && value !== internalInputValue ) {
		setInternalInputValue( value );
	}
```